content/700-optimize/400-recommendations/200-queries-on-unindexed-columns.mdx

```diff
@@ -0,0 +1,105 @@
+---
+title: 'Queries on unindexed columns'
+metaTitle: 'Optimize Recommendations: Queries on unindexed columns'
+metaDescription: "Learn about the recommendation provided by Optimize for queries on unindexed columns."
+tocDepth: 3
+toc: true
+---
+
+The following queries targeting the `user` model have a [`where` property](/orm/prisma-client/queries/filtering-and-sorting) for filtering on columns that do not have indexes.
+
+```ts
+await prisma.user.findFirst({
+   where: {
+      name: "Marc"
+   }
+})
+
+await prisma.user.findFirst({
+   where: {
+      name: "Jon"
+   }
+})
+
+await prisma.user.count({
+   where: {
+      name: "Nikolas"
+   }
+})
+```
+
+## Why is this a problem?
+
+An index allows the database to retrieve data more quickly. An index is similar to an index in a book: it helps you quickly locate the information you're looking for without having to go through every page.
+
+When providing Prisma with a `where` property, if there are no indexes defined for the columns provided, the database might need to look through every row in the table (a *”full table scan”*) to determine whether or not it matches the relevant conditions. This can be undesirable for several reasons:
```

Comment:
When using a `where` property in Prisma, if there are no indexes on the specified columns, the database may perform a *"full table scan"*—checking every row in the table to find matches. This can be undesirable for several reasons: